suite.go

```diff
@@ -248,6 +264,16 @@ func (s *Suite) Step(expr interface{}, stepFunc interface{}) {
 	s.steps = append(s.steps, def)
 }
 
+// GetGherkinFeatures returns parsed Gherkin features for the suite
+// It can be useful to retrieve some information about Gehrkin document, Pickles ...
+func (s *Suite) GetGherkinFeatures() []GherkinFeature {
```

Comment:
What do you think about instead of this function returning interfaces, it would return structs which includes the GherkinDocument and the Pickles?

I'm not too happy about returning pointers, sense that would mean that the test could modify the data, I would like to keep potential side effects to a minimum.

Reply:
Sometimes the developer writing the tests may want to modify some data before the test. This interface way would be a good way to give this possiblity. As it stays in the suite, it would be done before the whole testing.
Or would it make to much side effects ?

But I can change the GherkinFeature to a struct instead right with 2 attributes for pickles and gherkindocument

Reply:
Shouldn't it be better to make the interface but add some explicit way to modify data so we can assume the developer now what he is doing if something break ?

Reply:
I would say it is a bit too much, and would add more maintenance ...
but a comment on the interface warning about modifications could do or ?
@lonnblad what do you think ?

Reply:
LGTM. @lonnblad ?